File: 392-is-subsequence/is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int slow = 0;

        for(int fast = 0; fast < t.size(); fast++){
            if(t[fast]==s[slow]){
                slow++;
            }
        }
        if(slow == s.size()){
            return true;
        }
        else{
            return false;
        }



        
    }
};

File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        slow = 0

        for fast in range(len(t)):
            if slow < len(s) and t[fast] == s[slow]: #slow <len(s) is to handle a special case, the other condition is logic
                slow += 1
        
        if slow == len(s):
            return True
        else:
            return False
        